### PR TITLE
feat: add documentation

### PR DESCRIPTION
**Pull request by kassianh**

### CODE REVIEW

File: README.tfdoc.hcl [hunk 1]
```diff
@@ -0,0 +1,813 @@
...
+  }
+
+  section {
+    title   = "Getting Started"
+    content = <<-END
+      Most common usage of the module:
+
+      ```hcl
+      module "terraform-aws-ecs-service" {
+        source = "git@github.com:mineiros-io/terraform-aws-ecs-service.git?ref=v0.0.1"
```
Comment (soerenmartius) [inline README.tfdoc.hcl:69]: `cluster` and `task_definition` are missing here which are both variables that are required for a working example. For an example please see the existing tests

**Review by soerenmartius — CHANGES_REQUESTED** [review]

File: README.tfdoc.hcl (hunk 1 above)
Comment (kassianh) [reply README.tfdoc.hcl:69]: Good catch, thanks!

File: README.tfdoc.hcl [hunk 2]
```diff
@@ -0,0 +1,826 @@
...
+          type        = bool
+          description = <<-END
+            Assign a public IP address to the ENI (Fargate launch type only).
+            Valid values are `true` or `false`.
+          END
+        }
+      }
+
+      variable "capacity_provider_strategy" {
+        type        = set(capacity_provider_strategy)
```
Comment (soerenmartius) [inline README.tfdoc.hcl:162]:
```suggestion
        type        = list(capacity_provider_strategy)
```

This is a list afaik

File: README.tfdoc.hcl [hunk 3]
```diff
@@ -0,0 +1,826 @@
...
+            attribute that is applied to a container instance. For the `binpack`
+            type, valid values are `memory` and `cpu`. For the `random` type, this
+            attribute is not needed. For more information, see
+            [Placement Strategy](https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_PlacementStrategy.html).
+          END
+        }
+      }
+
+      variable "placement_constraints" {
+        type        = set(placement_constraint)
```
Comment (soerenmartius) [inline README.tfdoc.hcl:344]:
```suggestion
        type        = list(placement_constraint)
```

this is a list afaik

File: README.tfdoc.hcl [hunk 4]
```diff
@@ -0,0 +1,826 @@
...
+            Cluster Query Language expression to apply to the constraint. Does
+            not need to be specified for the `distinctInstance` type. For more
+            information, see [Cluster Query Language](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/cluster-query-language.html)
+            in the Amazon EC2 Container Service Developer Guide.
+          END
+        }
+      }
+
+      variable "load_balancers" {
+        type        = set(load_balancer)
```
Comment (soerenmartius) [inline README.tfdoc.hcl:374]:
```suggestion
        type        = list(load_balancer)
```

this is a list afaik

**Review by soerenmartius — CHANGES_REQUESTED** [review]

File: README.tfdoc.hcl (hunk 2 above)
Comment (kassianh) [reply README.tfdoc.hcl:162]: @soerenmartius The `schema.json` has nesting mode as `set`:
```
"block_types": {
                            "capacity_provider_strategy": {
                                "nesting_mode": "set",
                                "block": {
                                    "attributes": {
                                        "base": {
                                            "type": "number",
                                            "description_kind": "plain",
                                            "optional": true
                                        },
                                        "capacity_provider": {
                                            "type": "string",
                                            "description_kind": "plain",
                                            "required": true
                                        },
                                        "weight": {
                                            "type": "number",
                                            "description_kind": "plain",
                                            "optional": true
                                        }
                                    },
                                    "description_kind": "plain"
                                }
                            },
```

File: README.tfdoc.hcl (hunk 3 above)
Comment (kassianh) [reply README.tfdoc.hcl:344]: Same with this:
```
"placement_constraints": {
                                "nesting_mode": "set",
                                "block": {
```

File: README.tfdoc.hcl (hunk 4 above)
Comment (kassianh) [reply README.tfdoc.hcl:374]: And this:
```
"load_balancer": {
                                "nesting_mode": "set",
```

File: README.tfdoc.hcl (hunk 2 above)
Comment (soerenmartius) [reply README.tfdoc.hcl:162]: gotcha thanks!

**Review by soerenmartius — APPROVED** [review]
lgtm, please merge on your own behalf!

**Pull request merged**